R/as-data-frame.R

```diff
@@ -73,3 +102,96 @@ as.data.frame.CrunchDataFrame <- function (x, row.names = NULL, optional = FALSE
     names(out) <- names(x)
     return(structure(out, class="data.frame", row.names=c(NA, -nrow(x))))
 }
+
+
+#' Merge a CrunchDataFrame
+#' 
+#' Merging a CrunchDataFrame with a local dataframe is experiemental and might 
+#' result in unexpected results. One known issue is that using `merge` on a 
+#' CrunchDataFrame will change the both the CrunchDataFrame used as input as 
+#' well as create a new CrunchDataFrame. 
+#' 
+#' `merge`ing a CrunchDataFrame with a local dataframe is useful in situations 
+#' where you have new information in your local R session that you want to 
+#' connect with Crunch data. For example, this is especially useful for making
+#' plots with crunch and non-crunch data. It produces a hybrid CrunchDataFrame
+#' that has the local data attached to it, but like normal CrunchDataFrames 
+#' it is still judicious about downloading data from the server only when it 
+#' is needed.
+#' 
+#' @param x a CrunchDataFrame
+#' @param y a standard data.frame
+#' @param by.x name of the variable to match
+#' @param by.y name of the variable to match
```

Comment:
I tried for a while to unify the functions used for joining datasets, but in the end I found that having the helper function locate the by variables returning the values of the variable was working against me more than it was helping. I've since abstracted a bit, and put the checking and processing the `by`s for either `data.frame`s or `CrunchDataFrame`s into a separate function similar to how join works, but separate to not impact the join logic. Let me know if this is a big problem, and I can see if there is a way to unify them (possibly involving changing the way join works and what it expects from that helper function.)